Tests/XcodeProjTests/Objects/Project/PBXProjIntegrationTests.swift

```diff
@@ -14,6 +14,28 @@ final class PBXProjIntegrationTests: XCTestCase {
         }
     }
 
+    func test_uuids_after_adding_to_build_phase() throws {
+        let fixturePath = self.fixturePath().parent().parent()
+        print(fixturePath)
+        let xcp = try XcodeProj(path: self.fixturePath().parent())
+        let project = xcp.pbxproj.projects.first(where: { $0.name == "Project" })!
+        let iosGroup = project.mainGroup.children.first(where: { $0.path == "iOS" }) as! PBXGroup
+        let appTarget = xcp.pbxproj.nativeTargets.first(where: { $0.name == "iOS" })!
+        let sourcesPhase = appTarget.buildPhases.first(where: { $0.buildPhase == .sources }) as! PBXSourcesBuildPhase
+
+        for filename in ["FileNotInProject.swift"] {
+            let newFilePath = Path(components: fixturePath.components + ["iOS", filename])
+            let file = try iosGroup.addFile(at: newFilePath, sourceRoot: fixturePath)
+            let buildFile = PBXBuildFile(file: file, product: nil, settings: [:])
+            sourcesPhase.files?.append(buildFile)
+        }
```

Comment:
When the loop body exits, `buildFile` and `file` are deallocated, so the weak references pointing to them are set to nil. The "fix" would be to add them to a list.

It was not at all clear to me reading the docs that I need to maintain strong references to objects I create before writing output.